### PR TITLE
chore(deps): upgrade everything to latest Next and React

### PR DESCRIPTION
**Pull request by ascorbic**
### Summary

Manually upgrades demo sites and packages to Next 13.0.3 and React 18.2.0. Fixes a unit test broken by this.

### Test plan

Check it all builds ok

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [ ] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

**Comment by ascorbic** [comment]
Something is up with the API route dependency resolution

File: demos/next-auth/netlify.toml [hunk 1]
```diff
@@ -5,7 +5,6 @@ ignore = "if [ $CACHED_COMMIT_REF == $COMMIT_REF ]; then (exit 1); else git diff
 
 [build.environment]
 TERM = "xterm"
-NODE_VERSION = "17"
```
Comment (nickytonline) [inline demos/next-auth/netlify.toml:8]: Is this safe to remove? I can't remember why we had this set.

File: demos/next-export/netlify.toml [hunk 2]
```diff
@@ -5,7 +5,6 @@ ignore = "if [ $CACHED_COMMIT_REF == $COMMIT_REF ]; then (exit 1); else git diff
 
 [build.environment]
 NETLIFY_NEXT_PLUGIN_SKIP = "true"
-NODE_VERSION = "17"
```
Comment (nickytonline) [inline demos/next-export/netlify.toml:8]: Same as above

**Review by sarahetter — APPROVED** [review]

File: packages/next/test/request.spec.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import Chance from 'chance'
 import { NextURL } from 'next/dist/server/web/next-url'
-import { NextCookies } from 'next/dist/server/web/spec-extension/cookies'
+import { RequestCookies } from 'next/dist/server/web/spec-extension/cookies'
```
Comment (nickytonline) [inline packages/next/test/request.spec.ts:3]: Is this to fixe the `RespsonseCookies.getAll()`?

File: packages/runtime/src/helpers/files.ts [hunk 4]
```diff
@@ -356,7 +356,7 @@ export const getDependenciesOfFile = async (file: string) => {
     return []
   }
   const dependencies = await readJson(nft, 'utf8')
-  return dependencies.files.map((dep) => resolve(file, dep))
+  return dependencies.files.map((dep) => resolve(dirname(file), dep))
```
Comment (nickytonline) [inline packages/runtime/src/helpers/files.ts:359]: This fixes the tests, i.e. snapshots below, but do we have something in place to test this aside from the snapshot?

File: demos/next-auth/netlify.toml (hunk 1 above)
Comment (ascorbic) [reply demos/next-auth/netlify.toml:8]: It was for Cypress. It caused breakages with this update, as there are packages with `engines` fields that don't like it. It didn't seem to be needed anymore, and the tests still pass.

File: packages/next/test/request.spec.ts (hunk 3 above)
Comment (ascorbic) [reply packages/next/test/request.spec.ts:3]: @nickytonline No, this is related to changes in Next 13, where they switched from a single `NextCookies` class into separate `RequestCookies` and `ResponseCookies` classes

File: packages/runtime/src/helpers/files.ts (hunk 4 above)
Comment (ascorbic) [reply packages/runtime/src/helpers/files.ts:359]: @nickytonline I added tests in #1780, which is a more comprehensive fix

**Pull request merged**